### ISSUE TITLE
How to minimise False positive? 

### ISSUE THREAD

**Issue by akashAD98** [ultralytics/yolov5#4194]
## ❔Question
I have trained my model & its performing well on validation data, i got 82% MAP(@0.5), But when i tested my model on frames/video its giving few false predictions.

My dataset has 25 classes & each class has more than 2000 instances.

I have classes like blood,cigarette ,pipe,tank..etc  

This are the few results

1. for red color  object its predicting blood, 
![bloodof](https://user-images.githubusercontent.com/62583018/127321279-19e3cdb5-ca2a-4b82-b976-926409b7c01e.png)

![humnbleddibn-car](https://user-images.githubusercontent.com/62583018/127321426-93ebe9c3-0026-47b4-b85f-b2b35449c536.png)

2.I have class of alcohol & its detecting god but for For any bottle or bottlelike shape object its predicting Alcohol

![bottler-alco](https://user-images.githubusercontent.com/62583018/127321402-83c30951-ffc5-4020-8d6f-4ab1d79d1dab.png)

3.I have a class cigarette & it's detecting an object which looks like cigarette.
![cig](https://user-images.githubusercontent.com/62583018/127321555-1b832559-0594-452d-94bc-5f1713c947c1.pn

@glenn-jocher 
i already read the 
Thanks. Im  looking for your suggetion

**Comment by adrianholovaty** [comment]
Try adding some images that have zero objects, as described in "background images" on [this page](https://docs.ultralytics.com/guides/model-training-tips/).

**Comment by akashAD98** [comment]
@adrianholovaty thanks for the reply,
I have 2 question 
1. This background image consists of all class objects with no annotations ?? can you give 1 example of it ?? or its any kind of image which has zero objects (objects not present in my class ) ? or is its simple plan baground
2. Apart from this there is no other way?

**Comment by glenn-jocher** [comment]
@akashAD98 👋 Hello! Thanks for asking about improving YOLOv5 🚀 training results. 

Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥ 1500 images per class recommended
- **Instances per class.** ≥ 10000 instances (labeled objects) per class recommended
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total). No labels are required for background images.

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. VOC, VisDrone, GlobalWheat). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
                                             custom_pretrained.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. COCO, Objects365, OIv6). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyp.scratch.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains:
http://karpathy.github.io/2019/04/25/recipe/

**Comment by akashAD98** [comment]
@adrianholovaty @glenn-jocher  so background images are images with plan background??  how to determine what kind of background images needs to add according to use case?
Note: please have look at my FP results. should i need to add this FP backround or its normal baground?
it would be great if you give 1 exmple.
Thank you so much!


I have 25 classes which are shotgun, pistol, rifle... but it's falling in real-world scenarios.
doing too much false prediction. so what are the best background images for this use case???




![shotgun](https://user-images.githubusercontent.com/62583018/127446636-4785d820-fa05-44fc-a696-d23df702d775.png)

**Comment by glenn-jocher** [comment]
@akashAD98 this is pretty simple. Images without objects are background images. Find them whereever you want.

**Comment by akashAD98** [comment]
@glenn-jocher  got it .thanks

**Comment by glenn-jocher** [comment]
@akashAD98 the only criteria is of course the images shouldn't contain the classes you are training on, so you could include the image shown above as a background image (provided of course it's not in your val set).

**Comment by majnas** [comment]
So, for a background image AAA.jpg we have to have an empty AAA.txt or no txt file is needed?

**Comment by glenn-jocher** [comment]
@m-nasiri this is already answered in https://github.com/ultralytics/yolov5/issues/4194#issuecomment-888280050
